src/modm/math/matrix.hpp

```diff
@@ -48,33 +47,40 @@ namespace modm
 	 * \author	Niklas Hauser
 	 * \author	Fabian Greif
 	 */
-	template<typename T, uint8_t ROWS, uint8_t COLUMNS>
+	template<typename T, std::size_t ROWS, std::size_t COLUMNS>
 	class Matrix
 	{
 	public:
+		static constexpr std::size_t RowCount = ROWS;
+		static constexpr std::size_t ColumnCount = COLUMNS;
+		static constexpr std::size_t ElementCount = RowCount * ColumnCount;
+
+		T element[ElementCount];
```

Comment:
The data has to be initialized for the `constexpr` default constructor to be really `constexpr`.
```suggestion
		T element[ElementCount]{};
```